apps/dashboard/app/apps/user_preferences.rb

```diff
@@ -0,0 +1,43 @@
+class UserPreferences
+    require 'yaml'
+  
+    def initialize()
+      _prefs
+    end
+  
+    def preferences_file
+        @preferences_file ||= begin
+            file = ENV['OOD_PREFERENCES_FILE'] || "#{User.new.dir}/preferences.yml"
+            Pathname.new(file.to_s).expand_path
+        end
+    end
+    
+    private def _prefs
+        @_prefs ||= begin
+            if preferences_file.file? && preferences_file.readable?
+                YAML.safe_load(preferences_file.read)
+            else
+                {}
+            end
+        rescue => e
+            puts "#{e.message}"
+            @_prefs = {}
+        end
+    end
+  
+    def method_missing(method_name, *args, &block)
+        method_string = method_name.to_s
+        if @_prefs.key?(method_string)
+            @_prefs.dig(method_string, *args)
+        else
+            super
+        end
+    end
+
+    def write_setting(app_name, settings_hash)
+        settings_hash.each do |key, value|
+            @_prefs[app_name][key] = value
+        end
+        File.open(@preferences_file, "w") { |file| file.write(@_prefs.to_yaml) }
```

Comment:
I want to say this is all we need (with a rescue block here too in case it fails).

I'm not sure the `app_name` prefix here is super meaninful. It _was_ something I was playing around with in ood_core, but in the dashboard it has less meaning. Besides, let's just keep it simple and dump the whole thing.